### PR TITLE
CLN: Drop the as_recarray parameter in read_csv

### PR DESCRIPTION
**Pull request by gfyoung**
Deprecated back in 0.19.0

xref #13373.

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by gfyoung** [comment]
Everything is green now.  If there are no objections, I'll merge later today.

**Comment by jreback** [comment]
pls wait for my review

File: pandas/_libs/parsers.pyx [hunk 1]
```diff
@@ -2321,63 +2309,6 @@ cdef _apply_converter(object f, parser_t *parser, int64_t col,
...
-        length = len(columns.values()[0])
-
-    stride = dt.itemsize
-
-    # We own the data.
-    buf = <char*> malloc(length * stride)
-
-    recs = sarr_from_data(dt, length, buf)
-    assert(recs.flags.owndata)
-
```
Comment (jreback) [inline pandas/_libs/parsers.pyx]: i believe u can remove

sarr_from_data 
_fill_structured_columns 

from the codebase as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/parsers.pyx (hunk 1 above)
Comment (gfyoung) [reply pandas/_libs/parsers.pyx]: Why?  We return `recs`.

Comment (gfyoung) [reply pandas/_libs/parsers.pyx]: Hmm...potentially.

Comment (jreback) [reply pandas/_libs/parsers.pyx]: we don’t have flake8 on cython that looks for unused things fyi

Comment (gfyoung) [reply pandas/_libs/parsers.pyx]: Yeah...that's tougher to check, as `flake8` works on a file-by-file basis and has no holistic check on the code-base.

Comment (gfyoung) [reply pandas/_libs/parsers.pyx]: GitHub search seems to agree with your belief though.

**Comment by gfyoung** [comment]
@jreback : Those two functions have been removed.  PTAL.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @gfyoung

**Pull request merged**